packages/lesswrong/lib/notificationTypes.tsx

```diff
@@ -480,6 +480,31 @@ export const PostSharedWithUserNotification = registerNotificationType({
   }
 });
 
+export const PostAddedAsCoauthorNotification = registerNotificationType({
+  name: "addedAsCoauthor",
+  userSettingField: "notificationAddedAsCoauthor",
+  allowedChannels: ["onsite", "email", "both"],
+  async getMessage({documentType, documentId}: GetMessageProps) {
+    let document = await getDocument(documentType, documentId) as DbPost;
+    const name = await postGetAuthorName(document);
+    const postOrDialogue = document.collabEditorDialogue ? 'dialogue' : 'post';
+    return `${name} added you as a coauthor to the ${postOrDialogue} "${document.title}"`;
+  },
+  getIcon() {
+    return <Components.ForumIcon icon="Bell" style={iconStyles} />
+  },
+  getLink: ({documentType, documentId, extraData}: {
+    documentType: string|null,
+    documentId: string|null,
+    extraData: any
+  }): string => {
+    if (!documentId) {
+      throw new Error("PostAddedAsCoauthorNotification documentId is missing")
+    }
+    return postGetEditUrl(documentId, false)
```

Comment:
Personally I was surprised that this went to the edit link instead of the post page url, but nbd

Reply:
I think for being added as a coauthor to a dialogue, it's not super userful to get sent to the post page, since the intended interaction is to start participating in the dialogue